crates/sui-types/src/transaction.rs

```diff
@@ -635,6 +663,21 @@ impl ProgrammableTransaction {
             .collect())
     }
 
+    fn receiving_objects(&self) -> UserInputResult<Vec<ObjectRef>> {
```

Comment:
instead of returning a Result here, can we verify uniqueness in `validity_check` (below)?

Reply:
We could, although I would favor keeping it the way it is currently for two reasons:
1. It keeps the semantics of this function more in-line with the types of checks performed by the `input_objects` function above.
2. It keeps the semantic meaning of the `validity_check` function more in-line with what it does today (basically making sure that the various limits are respected).